Source/KSCrash/Source/KSCrash/Recording/BSG_KSCrashReport.c

```diff
@@ -1542,76 +1548,95 @@ void bsg_kscrashreport_writeStandardReport(
                           bsg_kscrw_i_addJSONData, &fd);
 
     writer->beginObject(writer, BSG_KSCrashField_Report);
-    { // TODO avoid collecting these fields for handled errors. they're not required at all.
-        bsg_kscrw_i_writeReportInfo(
+    {
+        bsg_kscrashreport_writeKSCrashFields(crashContext, writer);
```

Comment:
It would be cleaner to condition on the presence of `eventOverrides` to determine what is output.

The call to `bsg_kscrw_i_writeReportInfo` could be brought up to this level, then the calls to `bsg_kscrashreport_writeKSCrashFields`, `bsg_kscrashreport_writeBugsnagPayload` and `bsg_kscrashreport_writeOverrides` can each be conditioned on `crashContext->crash.userException.eventOverrides`. This would mean we wouldn't need to pass an extra parameter all the way down and it's only this function that needs to care what is written.

Reply:
Updated implementation to check for `eventOverrides` rather than passing a boolean parameter. This required changing the OOM notify call to pass in `nil` rather than a dictionary, as OOMs do require the KSCrash report fields.